tools/code_format/config.yaml

```diff
@@ -215,14 +218,22 @@ paths:
     - test/extensions/stats_sinks/wasm/wasm_stat_sink_test.cc
     - test/test_common/wasm_base.h
 
-dir_order:
-  - envoy
-  - common
-  - source
-  - exe
-  - server
-  - extensions
-  - test
+dir_order:                                                                     # unique
+  - envoy                                                                      # unique
+  - nighthawk                                                                  # unique
+  - external/source/envoy                                                      # unique
+  - external                                                                   # unique
```

Comment:
I am trying to verify if this list contains all the directories it should, but I am having a hard time figuring out how these map into the list of our actual directories here:

https://github.com/envoyproxy/nighthawk

This seems to be a list of top level and sub-level directories. Can you help me understand what this list should contain and how it is used?

E.g. the following directories are missing here, is that intentional?
- benchmarks (no c++ code, so probably ok)
- adaptive_load

Reply:
The short answer is I'm not 100% sure what this list's intention is. I copied this directly from:
   --include_dir_order envoy,nighthawk,external/source/envoy,external,api,common,source,exe,server,client,distributor,sink,grpcpp,request_source,test_common,test 

which did not contain adaptive_load or benchmarks.

My understanding of this list is that we probably have more things here than we actually need. Anything omitted I believe gets bundled into one section and alphabetized within that section. This list is ordered, so will ensure that (for example) external/source/envoy appears before anything else in external. I assume that for some envoy-related thing that header order matters, but for most files/subdirectories it shouldn't, and it would probably be fine to have them bundled in one section.

I don't remember what it is, but we've typically added directories to this list upon some mysterious error that we have encountered on occasion, but I forget what it is, or if it is still relevant after this yaml migration.

Reply:
Thank you, feel free to merge in assuming the files in the adaptive_load directories do get formatted even though they aren't listed.

Reply:
We talked about this offline, but after running an experiment, I realized that we are in fact missing some features in those directories. I am going to follow up separately adding directories and fixing issues